{
    "issue": {
        "title": "[Python] Legacy ParquetDataset gives confusing error message when passed a filter for non-directory dataset",
        "body": "***Note**: This issue was originally created as [ARROW-9780](https://issues.apache.org/jira/browse/ARROW-9780). Please see the [migration documentation](https://gist.github.com/toddfarmer/12aa88361532d21902818a6044fda4c3) for further details.*\n\n### Original Issue Description:\nNot highest priority since it's for the legacy dataset, but this is rather confusing:\r\n\r\n```Java\n\r\nIn [2]: pq.ParquetDataset(\"test.parquet\", filters=[(\"x\", \">\", 4)])   \r\n...\r\nAttributeError: 'NoneType' object has no attribute 'filter_accepts_partition'\r\n```\r\n\r\nBasically the filters argument only makes sense for a directory (because the legacy dataset only filters on partitions), but it shouldn't error like that for non-directories.\r\n\r\n\r\nFrom https://github.com/dask/dask/issues/6512",
        "created_at": "2020-08-18T07:20:09.000Z",
        "updated_at": "2022-09-23T14:11:45.000Z",
        "labels": [
            "Migrated from Jira",
            "Component: Python",
            "Type: enhancement"
        ],
        "closed": false
    },
    "comments": []
}